runtime/CMakeLists.txt

```diff
@@ -299,25 +299,34 @@ target_link_libraries(j9vm_main_wrapper
 # set up glue library target names
 # See https://github.com/eclipse/omr/blob/master/doc/BuildingWithCMake.md
 set(OMR_GC_GLUE_TARGET "j9vm_gc_glue" CACHE INTERNAL "")
+set(OMR_GC_GLUE_FULL_TARGET "j9vm_gc_glue_full" CACHE INTERNAL "")
 set(OMR_RAS_GLUE_TARGET "j9vm_ras_glue" CACHE INTERNAL "")
 set(OMR_CORE_GLUE_TARGET "j9vm_core_glue" CACHE INTERNAL "")
 set(OMR_UTIL_GLUE_TARGET "j9vm_util_glue" CACHE INTERNAL "")
 
 # setup OMR library overrides
 set(OMR_GC_LIB "j9gc" CACHE INTERNAL "")
+set(OMR_GC_FULL_LIB "j9gc_full" CACHE INTERNAL "")
 set(OMR_HOOK_LIB "j9hookable" CACHE INTERNAL "")
 set(OMR_PORT_LIB "j9prt" CACHE INTERNAL "")
 set(OMR_THREAD_LIB "j9thr" CACHE INTERNAL "")
 set(OMR_TRACE_LIB "j9trc" CACHE INTERNAL "")
 
 set(OMR_DDR_SET "j9ddr" CACHE INTERNAL "")
 
-add_subdirectory(gc_glue_java)
+# Declare the GC glue interface libraries that omr might reference,
+# including the 'full' version. We'll only add sources to the 'full'
+# version in gc_glue_java, below, when appropriate.
+j9vm_add_library(${OMR_GC_GLUE_TARGET} INTERFACE)
+j9vm_add_library(${OMR_GC_GLUE_FULL_TARGET} INTERFACE)
+
```

Comment:
is there a reason to use a forward declaration instead of just including gc_glue_java ?

Reply:
Yes, it's because the `_full` version of the glue library is only built if `OMR_MIXED_REFERENCES_MODE_STATIC=ON`, but `OMR_MIXED_REFERENCES_MODE_STATIC` is set as a cache variable later than when the glue libraries are referenced in other libraries. This results in having to run `make all` twice to get the build working, since on the second run, the cache variable is already set.

Reply:
Like the comment says:
```
# Declare the GC glue interface libraries that omr might reference,
# including the 'full' version. We'll only add sources to the 'full'
# version in gc_glue_java, below, when appropriate.
```

Reply:
we could just remove the conditional around gc_glue_java full. Interface libraries are never built on their own. So we can just define it but never reference it anywhere

Reply:
The point is that it must be _declared_ before including the omr.

Reply:
yes, but prior to this change we were including the gc_glue_java folder before including omr. My understanding is that the order needed to change because we gated the definition of `j9vm_gc_glue_full` based on `OMR_MIXED_REFERENCES_MODE_STATIC`, however `OMR_MIXED_REFERENCES_MODE_STATIC` will not be set until after omr is included. If instead of gating the definition, we always define `j9vm_gc_glue_full` we can continue to simply include the `gc_glue_java` folder before including omr

Reply:
No, the glue libraries must be declared before including omr, otherwise cmake will complain 'unknown target' at [1].

[1] https://github.com/sharon-wang/omr/blob/mixedbuild_cmake/gc/CMakeLists.txt#L536

Reply:
yes, but they are both declared, and defined in gc_glue_java, thus if we include gc_glue_java before including omr, they will be declared before omr is included (which is why the current builds are working)

Reply:
We don't want anything more than the (full) interface declaration in builds not using `OMR_MIXED_REFERENCES_MODE_STATIC`, but that  variable is defined by omr: we need to include omr before fleshing out those interfaces (if `OMR_MIXED_REFERENCES_MODE_STATIC` is set).

Reply:
I understand. My point is that you could just flush out both interfaces regardless. If you never link against the extra glue, it has no effect on the build system. With either solution, if for some reason you do link against the extra glue (with `OMR_MIXED_REFERENCES_MODE_STATIC` not set) you would get a link error at compile time.

Reply:
I think you have it backward: if we (unconditionally) provide complete library definitions, there would be no link error if the extra glue library were used (unintentionally).

Reply:
How do we want to proceed?

Reply:
I'm satisfied with this in it's current form, but I think this should wait until eclipse/omr#5657 is merged in case it evolves in a way that would alter how this handles `OMR_GC_POINTER_MODE == "mixed"`.

Reply:
Sounds good to me. @dnakamura What are your thoughts?